Sources/ArcGISToolkit/Components/FeatureTemplatePicker/FeatureTemplatePicker.swift

```diff
@@ -0,0 +1,334 @@
+            }
+        }
+        .contentShape(Rectangle())
+        .onTapGesture {
+            selection = info
+        }
+    }
+}
+
+extension FeatureTemplatePicker {
+    /// The model for the feature template picker.
+    @MainActor
+    final class Model: ObservableObject {
+        /// The associated geo model.
+        let geoModel: GeoModel
+        /// Include feature templates from tables where features cannot be created.
+        private let includeNonCreatableFeatureTemplates: Bool
+        /// Search text for filtering the list of templates.
+        var searchText: String = "" {
+            didSet { templatesOrSearchTextDidChange() }
+        }
+        /// The complete unfiltered list of feature template sections.
+        private var unfilteredFeatureTemplateSections = [FeatureTemplateSectionInfo]() {
+            didSet { templatesOrSearchTextDidChange() }
+        }
+        
+        /// The sections.
+        @Published fileprivate private(set) var featureTemplateSections = [FeatureTemplateSectionInfo]()
+        /// A Boolean value indicating whether templates are being generated.
+        @Published var isGeneratingFeatureTemplates = false
+        /// A Boolean value indicating if content is unavailable.
+        @Published var showContentUnavailable = false
+        /// A Boolean value indicating if no templates were found with the given search text.
+        @Published var showNoTemplatesFound = false
+        
+        /// Creates a model for a given geo model.
+        init(geoModel: GeoModel, includeNonCreatableFeatureTemplates: Bool) {
+            self.geoModel = geoModel
+            self.includeNonCreatableFeatureTemplates = includeNonCreatableFeatureTemplates
+        }
+        
+        /// Generates the feature templates.
+        func generateFeatureTemplates() async {
+            isGeneratingFeatureTemplates = true
+            defer { isGeneratingFeatureTemplates = false }
+            
+            unfilteredFeatureTemplateSections = await makeFeatureTemplateSections()
+        }
+        
+        /// The feature template section information for the geo model.
+        private func makeFeatureTemplateSections() async -> [FeatureTemplateSectionInfo] {
+            try? await geoModel.load()
+            var sections = [FeatureTemplateSectionInfo]()
+            let layersAndTables = geoModel.arcGISFeatureLayersAndTables
+            await layersAndTables.map(\.table).load()
+            for (layer, table) in layersAndTables {
+                guard includeNonCreatableFeatureTemplates || table.canAddFeature else { continue }
+                var infos = [FeatureTemplateInfo]()
+                for template in table.allTemplates {
+                    let feature = table.makeFeature(template: template)
+                    let symbol = layer.renderer?.symbol(for: feature)
+                    let image = try? await symbol?.makeSwatch(scale: UIScreen.main.scale)
```

Comment:
<img width="464" alt="Screenshot 2024-12-30 at 1 50 56 PM" src="https://github.com/user-attachments/assets/a5c988dd-0b36-4538-aba4-6b4ce37eef61" />

We'll need to either find an alternative or mark the component unavailable for Vision for now to avoid breaking the build.